Client/Frontend/Home/HistoryHighlights/HistoryHighlightsViewModel.swift

```diff
@@ -248,25 +212,23 @@ extension HistoryHighlightsViewModel: HomepageViewModelProtocol, FeatureFlaggabl
     }
 
     func numberOfItemsInSection(for traitCollection: UITraitCollection) -> Int {
-        guard let count = historyItems?.count else {  return 0 }
-
         // If there are less than or equal items to the max number of items allowed per column,
         // we can return the standard count, as we don't need to display filler cells.
         // However, if there's more items, filler cells needs to be accounted for, so sections
         // are always a multiple of the max number of items allowed per column.
-        if count <= UX.maxNumberOfItemsPerColumn {
-            return count
+        if historyItems.count <= UX.maxNumberOfItemsPerColumn {
+            return historyItems.count
         } else {
             return numberOfColumns * UX.maxNumberOfItemsPerColumn
         }
     }
 
     var hasData: Bool {
-        return !(historyItems?.isEmpty ?? true)
+        return !(historyItems.isEmpty)
     }
 
     func updateData(completion: @escaping () -> Void) {
-        loadItems(completion: completion)
+        historyItems = historyHighlightsDataAdaptor.getHistoryHightlights()
     }
```

Comment:
Should we call the completion after the `historyItems =` ?

Reply:
Mh, turns out I'll be removing completion in the clean up PR. You can disregard this comment